react-events-and-state/src/index.jsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isClicked: false };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState(prevState => ({
+      isClicked: !prevState.isClicked
+    }));
+  }
+  // -- or --
+  // handleClick() {
+  // this.setState({ isClicked: !this.state.isClicked });
```

Comment:
This is better, you almost never need to use the callback for `setState`, keep it simple and just pass an object with the updated value(s).